### DISCUSSION TITLE
Want to only display suggestions when there is a file in vault with that exact name

### DISCUSSION THREAD

**Discussion by mlevison** [tadashi-aikawa/obsidian-various-complements-plugin#260] [Q&A] [answered]
Background - my vault contains notes on over a thousand books.

Goal - I want autocomplete **only** on existing files. If I've only typed one word, then I only want one word suggestions. 

Is this possible?
Example a few minutes ago I started typing Innovation. Along with the Innovation note (good), I was offered a list of books that have Innovation their title. (Many). That get's distracting.

I tried playing with the options but don't fully understand their meaning.

**Comment by tadashi-aikawa** [comment]
@mlevison 
It would be helpful to tell me more detail. And could you tell me your settings copied by the [Copy plugin settings](https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/3.%20Commands/%F0%9F%8E%AE%20Copy%20plugin%20settings/) command?

**Reply by tadashi-aikawa** [reply]
If you have a note named 'Obsidian', would the following be what you expect?

| typed word | suggestions |
|-|-|
| O | none |
| Ob | none |
| Obs | none |
| Obsi | none |
| Obsid | none |
| Obsidi | none |
| Obsidia | none |
| Obsidian | "Obsidian" |

**Reply by avigilante** [reply]
@tadashi-aikawa - I'll jump in as I have the same ask as the OP, I believe: the tokens that the plugin is suggesting can be words or links. What if I only wanted it to suggest links instead?
So that if I have the word "Obsidian" in any note, but not a note called "Obsidian.md", no suggestion would be displayed, and if instead I do have the "Obsidian.md" note, that would be the only suggestion displayed.
Here are my settings, which I would have expected generate that exact behavior, but instead I'm still getting words suggestions, not just internal links.

<img width="712" alt="Screen Shot 2023-11-28 at 13 15 31" src="https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/assets/51001619/d4fb3b00-5db5-4a58-98a9-0c28f4c08c12">

PS: as I wrote on Discord, your plugin is the most essential one I use...I couldn't use Obsidian without it. THANKS!

**Reply by tadashi-aikawa** [reply]
@avigilante 
It seems like the settings are correct. Does that reproduce in the sandbox vault? If so, could you tell me the reproduction steps in the sandbox vault?

**Reply by avigilante** [reply, marked as answer]
Thanks @tadashi-aikawa (and thanks for replying on Discord too!). Yes, the settings worked. I just had to restart Obsidian (my bad, I didn't know).

**Comment by mlevison** [comment]
Sorry, this got lost. In my case my Obsidian vault has many notes that are book titles. Many of those start with "The ..." so any time I type the it offers a link. Ideally I would like it not to offer a link until at least 5 characters have matched.

**Reply by tadashi-aikawa** [reply]
@mlevison 
Does [⚙️ Min number of characters for trigger](https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/4.%20Options/4.1.%20Main/%E2%9A%99%EF%B8%8F%20Min%20number%20of%20characters%20for%20trigger/) resolve your issue?

**Reply by mlevison** [reply]
This is much improved - Danke
